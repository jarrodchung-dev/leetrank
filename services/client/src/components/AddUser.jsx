import React from "react";

const AddUser = (props) => {
  return (
    <>
      <h1 className="title">{props.title}</h1>
      <hr />
      <form className="form" onSubmit={props.handleSubmit}>
        <div className="field">
          <label className="label">Username</label>
          <div className="control">
            <input
              className="input"
              name="username"
              value={props.username}
              type="text"
              placeholder="i.e. johnsmith"
              onChange={props.handleChange}
              required
            />
          </div>
        </div>
        <div className="field">
          <label className="label">Email</label>
          <div className="control">
            <input
              className="input"
              name="email"
              value={props.email}
              type="email"
              placeholder="i.e. johnsmith@example.com"
              onChange={props.handleChange}
              required
            />
          </div>
        </div>
        <button className="button" type="submit">
          Submit
        </button>
      </form>
    </>
  );
};

export default AddUser;
