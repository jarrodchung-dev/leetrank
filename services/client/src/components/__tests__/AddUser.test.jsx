import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import AddUser from "../AddUser.jsx";

test("AddUser renders properly", () => {
  const wrapper = shallow(<AddUser />);
  const form = wrapper.find("form");
  expect(form.find("input").length).toBe(3);
  expect(form.find("input").get(0).props.name).toBe("username");
  expect(form.find("input").get(1).props.name).toBe("email");
  expect(form.find("input").get(2).props.type).toBe("submit");
});
