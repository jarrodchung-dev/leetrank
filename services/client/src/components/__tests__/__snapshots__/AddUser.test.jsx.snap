// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddUser /> should render a snapshot successfully 1`] = `
<form
  className="form"
>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Username
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        name="username"
        placeholder="i.e. johnsmith"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Email
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        name="email"
        placeholder="i.e. johnsmith@example.com"
        required={true}
        type="email"
      />
    </div>
  </div>
  <button
    className="button"
    type="button"
  >
    Submit
  </button>
</form>
`;
